src/embedded/embedded.c

```diff
@@ -270,17 +271,34 @@ char* monetdb_query(monetdb_connection conn, char* query, char execute, monetdb_
 		goto cleanup;
 	}
 
-	if (!m->results && m->rowcnt >= 0 && affected_rows) {
+	if (affected_rows) {
 		*affected_rows = m->rowcnt;
 	}
 
+#ifdef HAVE_EMBEDDED_JAVA
```

Comment:
Is this really specific to embedded Java? Can't this be done in a generic way?

Reply:
For the JDBC connection is tricky, but I can look further. Changing the monetdb_query function signature is probably a bad idea due to the large number of parameters.